pkg/querier/batch/chunk.go

```diff
@@ -43,23 +44,27 @@ func (i *chunkIterator) Seek(t int64, size int) bool {
 			i.batch.Index++
 		}
 		if i.batch.Index+size < i.batch.Length {
-			return true
+			return i.batch.ValueTypes
 		}
 	}
 
-	if i.it.FindAtOrAfter(model.Time(t)) {
-		i.batch = i.it.Batch(size)
-		return i.batch.Length > 0
+	if typ := i.it.FindAtOrAfter(model.Time(t)); typ != chunkenc.ValNone {
```

Comment:
This logic assumes that the chunk guarantees to always contain samples of 1 and only 1 type. Is that guaranteed? If yes, then we should comment about it in the `chunkIterator`.

Reply:
Will be skipping other value types in Batch since I reverted the Batch to only contain Floats so we enforce it now.
For the future: tsdb physical chunks can only contain a single type, but logical iterators may not.